code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -1089,8 +1089,9 @@
 				filtering = FALSE
 		if(filtering && moderator_internal.gases[filter_type])
 			var/datum/gas_mixture/removed = moderator_internal.remove_specific(filter_type, 20)
-			removed.temperature = moderator_internal.temperature
-			linked_output.airs[1].merge(removed)
+			if(removed)
+				removed.temperature = moderator_internal.temperature
+				linked_output.airs[1].merge(removed)
```

Comment:
`moderator_internal.gases[filter_type]` should really be `moderator_internal.gases[filter_type][MOLES]`. The temperature setting is unneeded, it should happen inside remove_specific itself

Reply:
Reverting this and using `moderator_internal.gases[filter_type][MOLES]` instead causes the following runtime:

> [15:46:59] Runtime in hypertorus.dm,1090: cannot read from list
>  proc name: fusion process (/obj/machinery/atmospherics/components/unary/hypertorus/core/proc/fusion_process)

Does this need to be chained? `if(filtering && moderator_internal.gases[filter_type] && moderator_internal.gases[filter_type][MOLES])`? Is there a cleaner way of achieving this?

Reply:
God damn it.

No you're right, checking the returned gasmix is cleaner. The temp thing really doesn't need to exist tho,

Reply:
Done